Kitodo/src/main/java/org/kitodo/production/forms/prozesskopie/TitleRecordLinkTab.java

```diff
@@ -0,0 +1,332 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.forms.prozesskopie;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale.LanguageRange;
+import java.util.Objects;
+
+import javax.faces.model.SelectItem;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kitodo.api.dataeditor.rulesetmanagement.RulesetManagementInterface;
+import org.kitodo.api.dataeditor.rulesetmanagement.StructuralElementViewInterface;
+import org.kitodo.api.dataformat.IncludedStructuralElement;
+import org.kitodo.api.dataformat.Workpiece;
+import org.kitodo.data.database.beans.Process;
+import org.kitodo.data.database.exceptions.DAOException;
+import org.kitodo.data.exceptions.DataException;
+import org.kitodo.production.dto.ProcessDTO;
+import org.kitodo.production.forms.ProzesskopieForm;
+import org.kitodo.production.helper.Helper;
+import org.kitodo.production.metadata.MetadataEditor;
+import org.kitodo.production.services.ServiceManager;
+import org.kitodo.production.services.data.ProcessService;
+import org.primefaces.model.DefaultTreeNode;
+import org.primefaces.model.TreeNode;
+
+/**
+ * Backing bean for the ‘title record link’ tab of the process creation dialog.
+ */
+public class TitleRecordLinkTab {
+    private static final Logger logger = LogManager.getLogger(TitleRecordLinkTab.class);
+
+    /**
+     * Maximum number of search hits to show.
+     */
+    private static final int MAXIMUM_NUMBER_OF_HITS = 10;
+
+    /**
+     * Process creation dialog to which this tab belongs.
+     */
+    private final ProzesskopieForm prozesskopieForm;
+
+    /**
+     * List of search hits to show.
+     */
+    private List<HitLink> hitLinks = Collections.emptyList();
+
+    /**
+     * Select items for the radio buttons to select the insertion position of
+     * the child link in the root element of the parent process.
+     */
+    private List<SelectItem> insertionPositionSelectionItems = Collections.emptyList();
```

Comment:
The JSF component is called `insertionPositionSelection` and this list contains the `items` of the component. See also in the intranet page “JSF Formularklassen”, under the point “Auswahllisten”.

Reply:
Please rename to `insertionPositions`!

Reply:
> 
> 
> The JSF component is called `insertionPositionSelection` and this list contains the `items` of the component. See also in the intranet page “JSF Formularklassen”, under the point “Auswahllisten”.
> I do not understand what you mean by type definition.

Sorry, I have never seen this page before and I would not have confirmed to that. Just because you made an Intranet page (which has apperently no one ever seen) doesn't make it a convention.

Not naming methods like elements in the front end page is common knowledge. How come none of the other developers is doing it like this? I'm not going to explain camelCase or something similar to you either.

Reply:
It may be that this happened during the time when you were not there. In any case, we had talked about it once. Also in the new implementation of the metadata editor I had stuck to it. Therefore, it irritated me that it should suddenly be wrong now. I changed it now.

Reply:
I also don't recall this fact and never have seen this page before so it looks I also was not there. From changes made by other developer I assume that they also were not there. In such a situation appears a question who actually, except you, was there.

Reply:
The naming concept was changed. This variable is now called `selectableInsertionPositions`.